p2p/simulations/http.go

```diff
@@ -455,9 +463,21 @@ func (s *Server) StreamNetworkEvents(w http.ResponseWriter, req *http.Request) {
 		}
 	}
 
+	//Only do throttling if requested
+	if req.URL.Query().Get("throttling") == "true" {
+		throttling = true
+		//rate at which we allow events to be emitted on the server-side event stream
+		rate := time.Second / 40
+		throttle = time.Tick(rate)
+	}
+
 	for {
 		select {
 		case event := <-events:
+			//if throttling, wait for the next throttle tick before emitting event
+			if throttling {
+				<-throttle
+			}
```

Comment:
Shouldn't this throttling code should be before the select? If there is no event the Tick will just flood the throttle channel without anything reading it.